src/rust/engine/options/src/parse_tests.rs

```diff
@@ -648,3 +661,190 @@ fn test_parse_heterogeneous_dict() {
         )
     );
 }
+
+#[test]
+fn test_expand_fromfile() {
+    let (_tmpdir, fromfile_pathbuf) = write_fromfile("fromfile.txt", "FOO");
+    let fromfile_path_str = format!("{}", fromfile_pathbuf.display());
+    assert_eq!(
+        Ok(Some(fromfile_path_str.clone())),
+        expand(fromfile_path_str.clone())
+    );
+    assert_eq!(
+        Ok(Some("FOO".to_string())),
+        expand(format!("@{}", fromfile_path_str))
+    );
+    assert_eq!(Ok(None), expand("@?/does/not/exist".to_string()));
```

Comment:
Could we also having a test for an optional file that _does_ exist, i.e. make sure that we're stripping off the `?` correctly? (I could imagine a future refactoring making a typo that causes us to pass in `?/does/not/exist` literally as the file path.)

In that vein, is it worth having "edge-case tests" that at least lock down the behaviour for "weird" things like `@@path/to/file` reading `@path/to/file` and similarly `@??path/to/file` (optionally) reading `?path/to/file`?

Reply:
Done. Note that `@@` is already covered - that is the escape hatch for a literal value that starts with `@`, so we never allow such a fromfile.